### PR TITLE
Document Query DSL and Highlight #334

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Review by rymsha — APPROVED** [review]

File: docs/api.adoc [hunk 1]
```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
...
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|fields: [String]!
+|List of fields (propertyPaths) to include in the search.
+
+|query: String!
+|A query string to match field value(s). Support the set of operators.
```
Comment (alansemenov) [inline docs/api.adoc:1469]:
```suggestion
|A query string to match field value(s). Supports the set of https://developer.enonic.com/docs/xp/next/storage/dsl#search_string_operators[operators].
```

File: docs/api.adoc [hunk 2]
```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
...
+
+|language: String!
+|A query string to match field value(s). Support the set of operators.
```
Comment (alansemenov) [inline docs/api.adoc:1472]:
```suggestion
|language: String!
|Content language that was used for stemming. List of https://developer.enonic.com/docs/xp/next/storage/indexing#languages[supported languages]
```

File: docs/api.adoc [hunk 3]
```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
...
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|fields: [String]!
+|List of fields (propertyPaths) to include in the search.
+
+|query: String!
+|A query string to match field value(s). Support the set of operators.
```
Comment (alansemenov) [inline docs/api.adoc:1495]: |query: String!
|A query string to match field value(s). Supports the set of https://developer.enonic.com/docs/xp/next/storage/dsl#search_string_operators[operators].

File: docs/api.adoc [hunk 4]
```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
...
+
+Exists DSL expression input type. Returns nodes that contain a value for a field.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|field: String!
+|Name of a field to check on existence.
```
Comment (alansemenov) [inline docs/api.adoc:1694]:
```suggestion
|Name of a field to check for existence.
```

File: docs/api.adoc [hunk 5]
```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
...
+
+Sort DSL expression input type.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|field: String!
+|Name of a field to check on existence.
```
Comment (alansemenov) [inline docs/api.adoc:1728]:
```suggestion
|Name of a field to sort by.
```

**Review by alansemenov — CHANGES_REQUESTED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**